python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,1093 @@
+
+    def copy(self, deep=False):
+        """Make a copy of the current instance
+        """
+        if deep:
+            filters = tuple([f for f in self.filters])
+            singles = tuple([self.singleType(s, deep=True) for s in self.singles])
+            result = type(self)(filters=filters, singles=singles)
+        else:
+            result = type(self)(image=self.image, mask=self.mask, variance=self.variance,
+                                filters=self.filters)
+        return result
+
+    def _slice(self, filters, filterIndex, indices):
+        """Slice the current object and return the result
+
+        See `Multiband._slice` for a list of the parameters.
+        """
+        image = self.image._slice(filters, filterIndex, indices)
+        mask = self._mask._slice(filters, filterIndex, indices)
+        variance = self._variance._slice(filters, filterIndex, indices)
+
+        result = type(self)(filters=filters, image=image, mask=mask, variance=variance)
+        assert np.all([r.getBBox() == result._bbox for r in [result._mask, result._variance]])
+        return result
+
+    def _verifyUpdate(self, image=None, mask=None, variance=None):
+        """Check that the new image, mask, or variance is valid
+
+        This basically means checking that the update to the
+        property matches the current bounding box and inherits
+        from the `MultibandBase` class.
+        """
+        for prop in [image, mask, variance]:
+            if prop is not None:
+                if prop.getBBox() != self.getBBox():
+                    raise ValueError("Bounding box does not match the current class")
+                if not isinstance(prop, MultibandBase):
+                    err = "image, mask, and variance should all inherit from the MultibandBase class"
+                    raise ValueError(err)
+
+    def getImage(self):
+        """Get the image
+        """
+        return self._image
+
+    def setImage(self, image):
+        """Set the image
+        """
+        self._verifyUpdate(image=image)
+        self._image = image
+        self._singles = self._buildSingles(image=image)
+
+    image = property(getImage, setImage)
+
+    def getMask(self):
+        """Get the mask
+        """
+        return self._mask
+
+    def setMask(self, mask):
+        """Set the mask
+        """
+        self._verifyUpdate(mask=mask)
+        self._mask = mask
+        self._singles = self._buildSingles(mask=mask)
+
+    mask = property(getMask, setMask)
+
+    def getVariance(self):
+        """Get the variance
+        """
+        return self._variance
+
+    def setVariance(self, variance):
+        """Set the variance
+        """
+        self._verifyUpdate(variance=variance)
+        self._variance = variance
+        self._singles = self._buildSingles(variance=variance)
+
+    variance = property(getVariance, setVariance)
+
+
+class MaskedPixel(object):
+    """A single pixel with an image, mask, and variance
+    """
```

Comment:
Agreed. We decided this after I had already written the classes but in case we found them useful I wanted to have them committed and removed by a later commit, I just forgot to remove them.